helm/cluster/values.schema.json

```diff
@@ -1955,6 +1976,51 @@
                     "title": "Provider",
                     "description": "The name of the Cluster API provider. The name here must match the name of the provider in cluster-<provider> app name."
                 },
+                "registry": {
+                    "title": "Registry",
+                    "description": "The registry to use for pulling images. If not set, the default registry is used.",
```

Comment:
Just a question for my understanding, why don't we use "default" as a key (similar to what we use for "kubernetesVersion") to use the registry `gsoci.azurecr.io` instead of allowing one of `templateName` vs `value`?

Reply:
Or just setting the `gsoci.azurecr.io` registry in {{ define "cluster.image.registry" }}

Reply:
So in our cluster-aws chart, we need to be able to provide the registry dynamically based on the region( atm based on `cn-` prefix in the region) so the customer or user doesn't need to provide a bunch of values for the cluster
 
 The only way how to do that is by using a template and providing the template name
, I added Also the `value` as this seems to be the common option that is used in `cluster` chart so wanted to keep that similar, but it is not used nor needed for our changes, so I would be fine to drop it. 

I didn't want to provide another default as the registry is internally already defined under `.values.internal.advancedConfiguration.registry` and so the changes either use what is set by the default there or use the value provided in `.providerIntegration`

Reply:
>  setting the gsoci.azurecr.io registry in {{ define "cluster.image.registry" }}


That was also in my head but I did not like that a default is hidden in a function/template somewhere

Reply:
Anyway what whatever makes you happy I can do, This just felt the right approach but I do not have that much experience with the `cluster` chart so it might not be the best, feel free to suggest better solution

Reply:
If we don't use the value I'd say lets drop it and leave the rest 👍🏻

Reply:
ACK , dropping

Reply:
Done